ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/SandboxIndexAndWriteService.scala

```diff
@@ -168,22 +179,42 @@ object SandboxIndexAndWriteService {
   }
 }
 
-class LedgerBackedWriteService(ledger: Ledger, timeProvider: TimeProvider) extends WriteService {
+final class LedgerBackedWriteService(ledger: Ledger, timeProvider: TimeProvider)(
+    implicit loggingContext: LoggingContext,
+) extends WriteService {
 
   override def currentHealth(): HealthStatus = ledger.currentHealth()
 
   override def submitTransaction(
       submitterInfo: ParticipantState.SubmitterInfo,
       transactionMeta: ParticipantState.TransactionMeta,
-      transaction: SubmittedTransaction): CompletionStage[ParticipantState.SubmissionResult] =
-    FutureConverters.toJava(ledger.publishTransaction(submitterInfo, transactionMeta, transaction))
+      transaction: SubmittedTransaction,
+  ): CompletionStage[ParticipantState.SubmissionResult] =
+    withEnrichedLoggingContext(
+      "submitter" -> submitterInfo.submitter,
+      "applicationId" -> submitterInfo.applicationId,
+      "commandId" -> submitterInfo.commandId,
+      "deduplicateUntil" -> submitterInfo.deduplicateUntil.toString,
+      "submissionTime" -> transactionMeta.submissionTime.toInstant.toString,
```

Comment:
And for that matter, `Instant`.

Reply:
I agree.

Reply:
Created https://github.com/digital-asset/daml/issues/6945